hbs-minifier-plugin.js

```diff
@@ -6,10 +6,19 @@ const stripWhiteSpace = Util.stripWhiteSpace;
 const isWhitespaceTextNode = Util.isWhitespaceTextNode;
 const hasLeadingOrTrailingWhiteSpace = Util.hasLeadingOrTrailingWhiteSpace;
 const stripNoMinifyBlocks = Util.stripNoMinifyBlocks;
+const canTrimBlockStatementContent = Util.canTrimBlockStatementContent;
+const canTrimElementNodeContent = Util.canTrimElementNodeContent;
+const canTrimModule = Util.canTrimModule;
+const assignDefaultValues = Util.assignDefaultValues;
+
+class BasePlugin {
+  constructor(env) {
+    env = env || {};
+    this.moduleName = env.moduleName;
```

Comment:
why do we need the `moduleName`?

Reply:
If a `component/partial(moduleName)` is specified, then we must not minify the entire template file.

Reply:
I don't think that will work reliably. If that component invokes another component you would also have to skip that one and so on.

Reply:
Yes. In that case,the current implementation requires all the components and partials used in main component needs to be mentioned in the config.

Reply:
this seems very error prone to me.

I'm okay with `components: ['foo-bar']` meaning that 
```hbs
{{#foo-bar}}
      x
{{/foo-bar}}
```  
does not get collapsed, but I don't think we should try to work across file boundaries if we can't be 100% sure to do the right thing.

Reply:
Ok. I'll make the necessary changes and update this PR.

Reply:
this seems unused now